html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/SMS.pm

```diff
@@ -91,10 +91,14 @@ Prompt fields with source specific SMS carriers
 sub prompt_fields {
     my ($self) = @_;
 
-    my @carriers = map { { label => $_->{name}, value => $_->{id} } } @{sms_carrier_view_all($self->source)};
-    $self->SUPER::prompt_fields({
-        sms_carriers => \@carriers, 
-    });
+    unless ( $self->source->type eq "Twilio" ) {
```

Comment:
You are right. I introduced an attribute "can_send_sms". Let's use it for that case too !